### PR TITLE
Remove PY3 checks in trial

### PR DESCRIPTION
**Pull request by rodrigc**
## Contributor Checklist:

* [X] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/9795
* [X] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [X] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [X] I have updated the automated tests.
* [X] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

File: src/twisted/trial/runner.py [hunk 1]
```diff
@@ -62,12 +61,8 @@ def isPackageDirectory(dirname):
     if dirname is a package directory.  Otherwise, returns False
     """
     def _getSuffixes():
-        if _PY3:
-            import importlib
-            return importlib.machinery.all_suffixes()
-        else:
-            import imp
-            return list(zip(*imp.get_suffixes()))[0]
+        import importlib
```
Comment (hawkowl) [inline src/twisted/trial/runner.py]: these two importlib imports can move to the top (1)

File: src/twisted/trial/runner.py [hunk 2]
```diff
@@ -126,20 +121,14 @@ def _importFromFile(fn, moduleName=None):
...
-            raise SyntaxError(fn)
-        module = importlib.util.module_from_spec(spec)
-        spec.loader.exec_module(module)
-        sys.modules[moduleName] = module
-    else:
-        import imp
-
-        with open(fn, 'r') as fd:
-            module = imp.load_source(moduleName, fn, fd)
+    import importlib
```
Comment (hawkowl) [inline src/twisted/trial/runner.py]: these two importlib imports can move to the top (2)

**Review by hawkowl — CHANGES_REQUESTED** [review]
Since there's no more reason for Py3TestLoader, the subclass and the overridden functions could be integrated back in and delete a bunch of old code there. We could do this in another ticket, but it feels like it might make sense here.

File: src/twisted/trial/runner.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/trial/runner.py]: done

File: src/twisted/trial/runner.py (hunk 2 above)
Comment (rodrigc) [reply src/twisted/trial/runner.py]: done

**Comment by rodrigc** [comment]
> Since there's no more reason for Py3TestLoader, the subclass and the overridden functions could be integrated back in and delete a bunch of old code there. We could do this in another ticket, but it feels like it might make sense here.

DONE

**Pull request merged**